src/librustc_hir/definitions.rs

```diff
@@ -342,12 +342,8 @@ impl Definitions {
     }
 
     #[inline]
-    pub fn as_local_hir_id(&self, def_id: DefId) -> Option<hir::HirId> {
-        if let Some(def_id) = def_id.as_local() {
-            Some(self.local_def_id_to_hir_id(def_id))
-        } else {
-            None
-        }
+    pub fn as_local_hir_id(&self, def_id: LocalDefId) -> hir::HirId {
+        self.local_def_id_to_hir_id(def_id)
```

Comment:
This is hilarious now. Not sure we should in this PR but at some point maybe we can change everything to `def_id_to_hir_id`? `def_id_to` is just 1 longer than `as_local` but clearer IMO.

Reply:
I agree. Would `local_def_id_to_hir_id` be even clearer?

Reply:
I'm not sure tbh. It's a bit long but it might be worth it.